controllers/datamovement_controller_test.go

```diff
@@ -60,23 +135,54 @@ var _ = Describe("Data Movement Test" /*Ordered, (Ginkgo v2)*/, func() {
 				},
 				UserId:  uint32(os.Geteuid()),
 				GroupId: uint32(os.Getegid()),
-				Cancel:  false,
+				Cancel:  dmCancel,
 			},
 		}
-	})
 
-	JustBeforeEach(func() {
+		cm = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      DMConfigMapName,
+				Namespace: DMConfigMapNamespace,
+				Labels: map[string]string{
+					testLabelKey: testLabel,
+				},
+			},
+			Data: cmData,
+		}
+
+		// Create CM and verify label
+		if createCm {
+			Expect(k8sClient.Create(context.TODO(), cm)).To(Succeed())
+			Eventually(func(g Gomega) string {
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(cm), cm)).To(Succeed())
+				return cm.Labels[testLabelKey]
+			}).Should(Equal(testLabel))
+		}
+
+		// Create DM and verify label
 		Expect(k8sClient.Create(context.TODO(), dm)).To(Succeed())
+		Eventually(func(g Gomega) string {
+			g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+			return dm.Labels[testLabelKey]
+		}).Should(Equal(testLabel))
+
 	})
 
-	JustAfterEach(func() {
-		Expect(k8sClient.Delete(context.TODO(), dm)).To(Succeed())
+	AfterEach(func() {
+		// Remove datamovement
 		Eventually(func() error {
+			k8sClient.Delete(context.TODO(), dm)
```

Comment:
Why was this put into the eventually block?

Reply:
An object can only be deleted once though. And that should always succeed (unless your test case set `createCm = false`)

Reply:
I'll move these out of the Eventually()